### PR TITLE
Improve error message for assertRegexpMatches()

### PR DESCRIPTION
**Pull request by rschlussel**
Include the regex and text you are trying to match in the error message

Inspired by the automation failure that had no helpful output.  
@cawallin

### CODE REVIEW

**Comment by cawallin** [comment]
looks good

File: tests/base_test_case.py [hunk 1]
```diff
@@ -135,6 +135,7 @@ def assertLazyMessage(self, msg_func, assert_function, *args, **kwargs):
     # equivalent to python 2.7's unittest.assertRegexpMatches()
     def assertRegexpMatches(
             self, text, expected_regexp, msg="Regexp didn't match"):
+        msg = '%s: %r not found in %r' % (msg, expected_regexp, text)
```
Comment (ebd2) [inline tests/base_test_case.py]: This is problematic for multi-line stuff. BaseMainCase._format_expected_actual might be borrow/adapt-able to this.

Comment (rschlussel) [inline tests/base_test_case.py]: Pushed already. But i'll fix it.

Comment (ebd2) [inline tests/base_test_case.py]: Thanks. Could you please fixe assertRaisesRegexp while you're in here? I think we can just replace the assertTrue(re.search(...)) with a call to assertRegexMaches. You can close SWARM-2005 too, by the way :-)

**Pull request merged**